File: springboot-todo/src/main/kotlin/com/example/todo/TaskService.kt
package com.example.todo

import org.springframework.stereotype.Service
import java.util.concurrent.atomic.AtomicInteger

@Service
class TaskService {
    private val tasks = mutableListOf<Task>()
    private val nextId = AtomicInteger(1)

    fun findAll(): List<Task> = tasks.toList()

    fun create(input: TaskInput): Task {
        val task = Task(
            id = nextId.getAndIncrement(),
            name = input.name,
            description = input.description
        )
        tasks.add(task)
        return task
    }

    fun delete(id: Int) {
        tasks.removeIf { it.id == id }
    }
}

File: nestjs-todo/src/task.service.ts
import { Injectable } from '@nestjs/common';
import { Task } from './task.interface';

@Injectable()
export class TaskService {
  private tasks: Task[] = [];
  private nextId = 1;

  findAll(): Task[] {
    return this.tasks;
  }

  create(name: string, description: string): Task {
    const task: Task = {
      id: this.nextId++,
      name,
      description,
    };
    this.tasks.push(task);
    return task;
  }

  delete(id: number): void {
    this.tasks = this.tasks.filter(task => task.id !== id);
  }
}
